portfolio/src/main/java/com/google/sps/servlets/MarkerServlet.java

```diff
@@ -0,0 +1,72 @@
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.lang.Iterable;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Returns an array of Markers in JSON form */
+@WebServlet("/markers")
+public class MarkerServlet extends HttpServlet {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    
+    /** Gets all Markers from the datastore and returns them as JSON */
+    @Override
+    public void doGet(final HttpServletRequest request, final HttpServletResponse response) throws IOException {
+        response.setContentType("application/json");
+        Query query = new Query("Marker");
+        PreparedQuery results = datastore.prepare(query);
+        List<Entity> markerList = results.asList(FetchOptions.Builder.withLimit(20));
+        String json = listToJson(markerList);
+        response.getWriter().println(json);
+    }
+
+    /**Convert all Marker entities to a json object */
+    private String listToJson(List<Entity> markers){
```

Comment:
Curious, why use strings and json objects, as opposed to maybe some type of custom class? I could see the latter being a bit more complex, but possibly easier to organize. Same question with comments.

Reply:
Hm, I guess I never thought of it because comments were really simple in the beginning and it seemed unnecessary. I also send the json as a response, I don't know how I would handle that if I made a new class because it seems I would still need to convert that to a json object. 

It does sound like a better option, I just don't know to avoid converting that to json

Reply:
To clarify, why do you need to convert back to json? Is this for sending it back?

Reply:
Yes, I use the json in script.js

Reply:
Just added a custom class for comments